### PR TITLE
Remove `six` dependency

### PR DESCRIPTION
**Pull request by findtopher**
## What changes are proposed in this pull request?

- remove `six` dependency from `fiftyone.core.odm.utils` and `fiftyone.core.storage`
    - we don't support python2 so we really don't need backwards compatibility
- use `numpy` less than v2.0.0 (I get segfaults when using numpy 2.0.0)
- add `pydicom`, `shapely`, and `tensorflow` to the tests requirements file
    - `pip install dist/*.whl && pip install -r requirements/test.txt && pytest tests/unittests` fails without these packages

## How is this patch tested? If it is not, please explain why.

in an empty virtualenv:
```
pip install -U pip setuptools build wheel
make clean
make python
pip install dist/*.whl
pip install -r requirements/test.txt
pytest tests/unittests
```

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [x] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **Chores**
	- Updated codebase for Python 3 compatibility by removing `six` library dependencies.
	- Added and updated necessary dependencies: 
	  - Added `numpy<2` to `requirements/common.txt`.
	  - Added `pydicom`, `shapely`, and `tensorflow` to `requirements/test.txt`.
	  - Updated `numpy` version constraint to `<2` in `setup.py`.

### CODE REVIEW

**Comment by findtopher** [comment]
corresponding [Teams PR](https://github.com/voxel51/fiftyone-teams/pull/765)

File: fiftyone/core/odm/utils.py [hunk 1]
```diff
@@ -353,7 +352,7 @@ def get_implied_field_kwargs(value, dynamic=False):
     if isinstance(value, numbers.Number):
         return {"ftype": fof.FloatField}
 
-    if isinstance(value, six.string_types):
+    if isinstance(value, str):
```
Comment (findtopher) [inline fiftyone/core/odm/utils.py:355]: https://six.readthedocs.io/#six.string_types

> Possible types for text data. This is [basestring()](https://docs.python.org/2/library/functions.html#basestring) in Python 2 and str in Python 3.

File: fiftyone/core/odm/utils.py [hunk 2]
```diff
@@ -110,7 +109,7 @@ def deserialize_value(value):
 
         return value
 
-    if isinstance(value, six.binary_type):
+    if isinstance(value, bytes):
```
Comment (findtopher) [inline fiftyone/core/odm/utils.py:112]: https://six.readthedocs.io/#six.binary_type

> Type for representing binary data. This is str in Python 2 and bytes in Python 3.

File: fiftyone/core/storage.py [hunk 3]
```diff
@@ -1010,8 +1009,8 @@ def _delete_file(filepath):
 
 
 def _to_bytes(val, encoding="utf-8"):
-    b = val.encode(encoding) if isinstance(val, six.text_type) else val
-    if not isinstance(b, six.binary_type):
+    b = val.encode(encoding) if isinstance(val, str) else val
```
Comment (findtopher) [inline fiftyone/core/storage.py:1012]: https://six.readthedocs.io/#six.text_type

> Type for representing (Unicode) textual data. This is [unicode()](https://docs.python.org/2/library/functions.html#unicode) in Python 2 and str in Python 3.

File: setup.py [hunk 4]
```diff
@@ -48,7 +48,7 @@ def get_version():
     "matplotlib",
     "mongoengine==0.24.2",
     "motor>=2.5",
-    "numpy",
+    "numpy<2",
```
[the remark this answers is not in the document]
Comment (benjaminpkane) [reply setup.py:51]: My take is our users are usually pretty opinionated about what numpy version they want, and I've only seen one indirect numpy 2.0 issue internally

If passing tests need numpy<2.0 we can keep in the requirements file

[the remark this answers is not in the document]
Comment (findtopher) [reply setup.py]:
```❯ pip freeze | grep numpy
numpy==2.0.0

╭─ ~/workspace/fiftyone/zoink-six topher/zoink-six *3```
results in:
```
tests/unittests/utils3d_tests.py Fatal Python error: Segmentation fault
```

with tensorflow missing from the venv

```
pip install -U 'numpy<2'
```

the tests complete, with failures due to missing tensorflow, so it feels like numpy is gonna be our weakness here...

**Review by benjaminpkane — APPROVED** [review]
LGTM. I think numpy 2.0 issues can be reassessed in a separate task

File: setup.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (brimoor) [reply setup.py]: +1 for not being opinionated about which `numpy` version we (FO) require in `setup.py`.

`tensorflow` is an optional dependency so any `tensorflow` <> `numpy` compability fights should be handled in arbitration outside of our `setup.py`.

**Pull request merged**